addons/account/models/res_currency.py

```diff
@@ -43,30 +43,25 @@ def _has_accounting_entries(self):
         return bool(self.env['account.move.line'].search_count(['|', ('currency_id', '=', self.id), ('company_currency_id', '=', self.id)]))
 
     @api.model
-    def _get_query_currency_table(self, options):
+    def _get_query_currency_table(self, company_ids, conversion_date):
         ''' Construct the currency table as a mapping company -> rate to convert the amount to the user's company
         currency in a multi-company/multi-currency environment.
         The currency_table is a small postgresql table construct with VALUES.
-        :param options: The report options.
+        :param company_ids: list of company ids
+        :param conversion_date: date, used to determine the the currency rate between the individual companies and the user's company
         :return:        The query representing the currency table.
         '''
 
+        companies = self.env['res.company'].browse(company_ids)
         user_company = self.env.company
-        user_currency = user_company.currency_id
-        if options.get('multi_company', False):
-            companies = self.env.companies
-            conversion_date = options['date']['date_to']
-            currency_rates = companies.mapped('currency_id')._get_rates(user_company, conversion_date)
-        else:
-            companies = user_company
-            currency_rates = {user_currency.id: 1.0}
+        currency_rates = companies.mapped('currency_id')._get_rates(user_company, conversion_date)
```

Comment:
Done. But would it be an option to "same" thing in `_get_rates`? (`if self == company.currency_id`) Compared to the query the cost of the additional `if` probably does not matter?

Reply:
Hmmm, not sure: the rate of the main currency is not especially 1 ; I think nothing ensures that (when set by hand, at least). I might be wrong with the current code ; I just know it used to be like that. So, _get_rates might return something else than one for the main currency of the company. 

On the other hand, if I'm right, it'd mean the computation of the currency table is buggy and needs to be divided by the rate of the main currency. I suggest we don't block this task with this and you investigate that separately later.

Reply:
Oh yes, I think you are right: it does not have to be 1

Reply:
As I said, I wont block this PR with this. Try to find a breaking case with non-1 domestic rates, and if you have one, we'll see how to patch it :)

Reply:
In my last comment just wanted to agree that my proposed change to `_get_rates` would not work in all (corner) cases.

It does not matter for `_get_query_currency_table` in the "single company" case though I think.
The `currency_rates` may technically be wrong but we do not output them; just the `conversion_rates`.
There we put the conversion rate `1.0` in any case (which it should be) since we just divide the number `currency_rates[<user currency id>]` by itself to determine this rate (in our case `1.0 / 1.0 `).

So it is all good I think (and better than with the additional query) :smile: 

Maybe for clarity the `companies == user_company` branches could cover the `conversion_rates` computation instead of just the `currency_rates` computation (so we could just skip the `currency_rates` in the single company case and put `conversion_rates` directly). But ultimately it works just fine now.